### DISCUSSION TITLE
Custom domains for Multi-tenant web app

### DISCUSSION THREAD

**Discussion by veewoovee** [supabase/supabase#4000] [Questions]
How can i use different domains for authentication in a single installation? 

Seperate databases for authentication for each tenant domain?

There is a better solution?

Is there a way to create multiple .env for each domain?

**Comment by grieblius** [comment]
As I understand, you can have different domains with the single Supabase instalation.
In Authentication settings, assign your main domain as  "Site URL".
Then you can add URL's with your other domains into "Redirect URLs" sections. Just the problem is, that you need to add exact URLs, to which you plan to redirect after login.

Then you need to use `redirectTo` option with signIn/signUp methods.
Docs: https://supabase.com/docs/reference/javascript/auth-signin

`redirectTo` should be full url, including domain. And with other than Site Url domain it will work only if the exact url is added to "Redirect URLs" section. E.g. if added https://other-custom.domain/ and you pass it with redirectTo - it should work.

Just I'm not sure is this feature official for multi-domain support or it just works. Here is more information:
https://github.com/supabase/supabase/discussions/600#discussioncomment-810640

I think it should work for basic case, but unfortunately I have a case where I need some dynamic urls.
E.g. in my case I am sending invite URL to user with unique key in url. For having good UX I want to redirect to that unique url user after signup or login.  But now I can redirect only to exact url's, which are defined in "Redirect URLs".

So it would be very good, that in "Redirect URLs"  would work one of these options:
- would be enough to define only base url with other domain, e.g. `https://other-domain`, like with "Site URL". And then any redirect path would work for that domain.
- when added redirect url, e.g. `https://other-domain/redirect`, then query (GET) parameters would be ignored. Then we could pass some unique ID's via query params. Currently, if I'm passing any query parameter, if I want redirect to work, I need to add url with exact query param key and value.

So this issue prevents me not only for implemented multi-tenant (multi domain) app, but also I can't fully test new features on Vercel's preview environments. 
Would appreciate some comments from Supabase team regarding this.. :)

UPDATE: 
Just found, that it should work by providing wildcard in the path, e.g. `https://mysite.com/users/*`:
https://github.com/supabase/supabase/discussions/600#discussioncomment-3303368
